ledger/eval.go

```diff
@@ -480,6 +485,31 @@ func (eval *BlockEvaluator) transaction(txn transactions.SignedTxn, ad transacti
 		if !txn.Txn.Group.IsZero() && !eval.proto.SupportTxGroups {
 			return fmt.Errorf("transaction groups not supported")
 		}
+
+		if !txn.Lsig.Blank() {
+			recs := make([]basics.BalanceRecord, len(txgroup))
+			for i := range recs {
+				var err error
+				recs[i], err = cow.Get(txgroup[i].Txn.Sender, true)
+				if err != nil {
+					return fmt.Errorf("transaction %v: cannot get sender record: %v", txn.ID(), err)
+				}
+			}
```

Comment:
This section would cause a series of database requests. These should be:
1. further optimized ( since the same sender could be involved in multiple txns )
2. bounded. The number of requests here should be limited. ( if we do it somehwhere else on the stack than I guess it's ok to have it as is. )